### ISSUE TITLE
@unknownRoute generate error

### ISSUE THREAD

**Issue by bounty1342** [Milad-Akarie/auto_route_library#67]
Hi, 
using 'unknownRoute' annotation like this :

``` dart
@unknownRoute
  UnderConstruction underConstruction;
```

generate this error :

```
[FINE] auto_route_generator:autoRouteGenerator on lib/core/routing/router.dart:Running AutoRouteGenerator
[SEVERE] auto_route_generator:autoRouteGenerator on lib/core/routing/router.dart:
Error running AutoRouteGenerator
Bad state: No element
dart:core                                                               List.first
package:auto_route_generator/src/router_config_resolver.dart 34:54      RouterConfigResolver.resolve
package:auto_route_generator/auto_route_generator.dart 29:12            AutoRouteGenerator.generateForAnnotatedElement
package:source_gen/src/generator_for_annotation.dart 53:30              GeneratorForAnnotation.generate
package:source_gen/src/builder.dart 298:35                              _generate
package:source_gen/src/builder.dart 79:15                               _Builder._generateForLibrary
package:source_gen/src/builder.dart 71:11                               _Builder.build
package:build                                                           runBuilder
package:build_runner_core/src/generate/build_impl.dart 486:19           _SingleBuild._runForInput.<fn>.<fn>.<fn>
package:build_runner_core/src/generate/performance_tracker.dart 302:15  _NoOpBuilderActionTracker.trackStage
package:build_runner_core/src/generate/build_impl.dart 484:23           _SingleBuild._runForInput.<fn>.<fn>
package:timing/src/timing.dart 222:44                                   NoOpTimeTracker.track
package:build_runner_core/src/generate/build_impl.dart 441:22           _SingleBuild._runForInput.<fn>
package:pool/pool.dart 127:28                                           Pool.withResource
package:build_runner_core/src/generate/build_impl.dart 437:17           _SingleBuild._runForInput
package:build_runner_core/src/generate/build_impl.dart 375:38           _SingleBuild._runBuilder.<fn>
dart:async                                                              Future.wait
package:build_runner_core/src/generate/build_impl.dart 374:36           _SingleBuild._runBuilder
dart:async                                                              _AsyncAwaitCompleter.start
package:build_runner_core/src/generate/build_impl.dart 372:40           _SingleBuild._runBuilder
package:build_runner_core/src/generate/build_impl.dart 320:20           _SingleBuild._runPhases.<fn>.<fn>
dart:async                                                              _completeOnAsyncReturn
package:build_runner_core/src/generate/build_impl.dart                  _SingleBuild._matchingPrimaryInputs
```

Regards

**Comment by Milad-Akarie** [comment]
@bounty1342 I'll look into it

**Comment by Milad-Akarie** [comment]
@bounty1342  you should get an exception that says unknownRoute page constructor should have one String parameter but something went wrong.. 
anyways it should work if you add the String parameter
```
class UnderConstruction extends StatelessWidget {
  const UnderConstruction(String routeName)
...
```

**Comment by Milad-Akarie** [comment]
fixed in auto_route_generator: 0.4.5
Thanks for reporting
